x-pack/plugins/security_solution/public/timelines/components/timeline/helpers.tsx

```diff
@@ -63,35 +63,30 @@ const buildQueryMatch = (
       : `${dataProvider.queryMatch.field} ${EXISTS_OPERATOR}`
   }`.trim();
 
-const buildQueryForAndProvider = (
-  dataAndProviders: DataProvidersAnd[],
-  browserFields: BrowserFields
-) =>
-  dataAndProviders
-    .reduce((andQuery, andDataProvider) => {
-      const prepend = (q: string) => `${q !== '' ? `${q} and ` : ''}`;
-      return andDataProvider.enabled
-        ? `${prepend(andQuery)} ${buildQueryMatch(andDataProvider, browserFields)}`
-        : andQuery;
-    }, '')
-    .trim();
-
 export const buildGlobalQuery = (dataProviders: DataProvider[], browserFields: BrowserFields) =>
   dataProviders
-    .reduce((query, dataProvider: DataProvider, i) => {
-      const prepend = (q: string) => `${q !== '' ? `${q} or ` : ''}`;
-      const openParen = i >= 0 && dataProviders.length > 1 ? '(' : '';
-      const closeParen = i >= 0 && dataProviders.length > 1 ? ')' : '';
-      return dataProvider.enabled
-        ? `${prepend(query)}${openParen}${buildQueryMatch(dataProvider, browserFields)}
-        ${
-          dataProvider.and.length > 0
-            ? ` and ${buildQueryForAndProvider(dataProvider.and, browserFields)}`
-            : ''
-        }${closeParen}`.trim()
-        : query;
-    }, '')
-    .trim();
+    .reduce((queries: string[], dataProvider: DataProvider) => {
+      const flatDataProviders = [dataProvider, ...dataProvider.and];
+      const activeDataProviders = flatDataProviders.filter(
+        (flatDataProvider) => flatDataProvider.enabled
+      );
+
+      if (!activeDataProviders.length) return queries;
+
+      const activeDataProvidersQueries = activeDataProviders.map((activeDataProvider) =>
+        buildQueryMatch(activeDataProvider, browserFields)
+      );
+
+      const activeDataProvidersQueryMatch = activeDataProvidersQueries.join(' and ');
+
+      return [...queries, `${activeDataProvidersQueryMatch}`];
```

Comment:
```suggestion
      return [...queries, activeDataProvidersQueryMatch];
```
You do not need to do that since it is already a string because of the join